### PR TITLE
add BaseTest, implement Guicy injection for CartWebPage

### PR DESCRIPTION
**Pull request by TarasTsyhanko**

### CODE REVIEW

File: src/main/java/modules/CartPageModule.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package modules;
+
+import com.google.inject.AbstractModule;
+import page.AbstractBasePage;
+import page.CartWebPage;
+
+public class CartPageModule  extends AbstractModule {
+    @Override
+    protected void configure() {
+        bind(CartWebPage.class);
```
Comment (mtsyhanko) [inline src/main/java/modules/CartPageModule.java:10]: create this for BasePage not for CartWebPage

File: src/test/java/tests/BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import modules.CartPageModule;
+import org.testng.annotations.AfterClass;
+
+import org.testng.annotations.BeforeClass;
+
+import page.CartWebPage;
+import validators.CartValidator;
+
+public class BaseTest {
+    protected CartActions cartActions;
```
Comment (mtsyhanko) [inline src/test/java/tests/BaseTest.java:16]: Why you need cart actions in BaseTest?

Comment (mtsyhanko) [reply src/test/java/tests/BaseTest.java:16]: The same here

File: src/main/java/bo/CartActions.java [hunk 3]
```diff
@@ -1,38 +1,40 @@
...
 import java.util.stream.Collectors;
 
 public class CartActions {
-    private final CartWebPage boxPage;
+    @Inject
+    private final CartWebPage cartPage;
 
-    public CartActions() {
-        boxPage = new CartWebPage();
+    public CartActions(CartWebPage cartWebPage) {
```
Comment (MykhailoTsiganko) [inline src/main/java/bo/CartActions.java]: Why do you need this Constructor

File: src/test/java/tests/BaseTest.java (hunk 2 above)
Comment (MykhailoTsiganko) [inline src/test/java/tests/BaseTest.java]: why do you need this two classes in BaseTest

File: src/test/java/tests/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    protected CartValidator validator;
+
+    @BeforeClass
+    public void setUp() {
+        Injector injector = Guice.createInjector(new CartPageModule());
+        CartWebPage cartPage = injector.getInstance(CartWebPage.class);
```
Comment (MykhailoTsiganko) [inline src/test/java/tests/BaseTest.java]: Why do you need this line. Remove

**Review by MykhailoTsiganko — CHANGES_REQUESTED** [review]

File: src/test/java/tests/BaseTest.java (hunk 2 above)
Comment (TarasTsyhanko) [reply src/test/java/tests/BaseTest.java]: this classes I should create in CartTest?

File: src/main/java/bo/CartActions.java (hunk 3 above)
Comment (TarasTsyhanko) [reply src/main/java/bo/CartActions.java]: becouse I should initisialize cartPage

File: src/test/java/tests/BaseTest.java (hunk 2 above)
Comment (MykhailoTsiganko) [reply src/test/java/tests/BaseTest.java]: but not in the BaseTest. You should use @Inject for Actions and Validators as well as for Pages.
Don't use 'new'

File: src/main/java/bo/CartActions.java (hunk 3 above)
Comment (MykhailoTsiganko) [reply src/main/java/bo/CartActions.java]: you dount need constructor at all. Pages should be initializated without constructors

**Review by MykhailoTsiganko — CHANGES_REQUESTED** [review]

File: src/test/java/tests/BaseTest.java (hunk 2 above)
Comment (TarasTsyhanko) [reply src/test/java/tests/BaseTest.java]: okey, where I should inject Page , in Actions, 
and Actions ,Validators I inject in Test  ?

Comment (TarasTsyhanko) [reply src/test/java/tests/BaseTest.java]: fixed

File: src/test/java/tests/BaseTest.java (hunk 4 above)
Comment (TarasTsyhanko) [reply src/test/java/tests/BaseTest.java]: fixed

File: src/main/java/bo/CartActions.java (hunk 3 above)
Comment (TarasTsyhanko) [reply src/main/java/bo/CartActions.java]: fixed

File: src/test/java/tests/GoodsInCartTest.java [hunk 5]
```diff
@@ -1,37 +1,35 @@
...
+public class GoodsInCartTest extends BaseTest {
     private CartActions cartActions;
     private CartValidator validator;
 
     @BeforeMethod
-    public void setUp(){
-        cartActions = new CartActions();
-        validator = new CartValidator();
+    public void initialize() {
+        injector = Guice.createInjector(new CartPageModule());
```
Comment (MykhailoTsiganko) [inline src/test/java/tests/GoodsInCartTest.java]: Why do you need this?
Cant you just method

all you need to write is
@Inject
private CartActions cartActions;

**Review by MykhailoTsiganko — CHANGES_REQUESTED** [review]

File: src/test/java/tests/GoodsInCartTest.java (hunk 5 above)
Comment (TarasTsyhanko) [reply src/test/java/tests/GoodsInCartTest.java]: without this method I have NullPointerExeptions

**Review by MykhailoTsiganko — APPROVED** [review]

**Pull request merged**